pallets/payment/src/lib.rs

```diff
@@ -93,10 +91,11 @@ pub mod pallet {
 			recipient: T::AccountId,
 			asset: AssetIdOf<T>,
 			amount: BalanceOf<T>,
+			resolver: Option<T::AccountId>,
```

Comment:
Nope, this parameter should go away. I don't think neither the person paying or getting payed should be able to influence in anyway what account has the permission to resolve the disputes since you could select a friend that is biased to vote in your favor.